src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/QueryParameterValueResolver.php

```diff
@@ -81,6 +90,22 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
             throw new NotFoundHttpException(sprintf('Invalid query parameter "%s".', $name));
         }
 
+        if (null !== $enumType && null !== $value) {
+            $enumFrom = static function ($value) use ($type, $name) {
+                if (null !== $value && !\is_string($value) && !\is_int($value)) {
+                    throw new NotFoundHttpException(sprintf('Invalid query parameter "%s": expecting an int or string, got "%s".', $name, get_debug_type($value)));
+                }
+
+                try {
+                    return null === $value ? null : $type::from($value);
+                } catch (\ValueError $e) {
+                    throw new NotFoundHttpException(sprintf('Invalid query parameter "%s": '.$e->getMessage().'.', $name), $e);
```

Comment:
I realized that we're missing a check linke this one (and a corresponding test case): 
```suggestion
                    if ($attribute->flags & \FILTER_NULL_ON_FAILURE) {
                        return null;
                    }
                    throw new NotFoundHttpException(sprintf('Invalid query parameter "%s": '.$e->getMessage().'.', $name), $e);
```

Reply:
Up to finish this aspect @andersmateusz?

Reply:
Ok